### PR TITLE
RI-7632: initial implementation of v2 api client

### PR DESCRIPTION
**Pull request by ArtemHoruzhenko**
## What

This PR implements the initial support for RDI API v2 client with automatic version detection and graceful fallback to v1.

#### 1. **ApiV2RdiClient** - New client implementation for RDI API v2
- Extends `ApiRdiClient` (v1) to inherit existing functionality
- Implements `getInfo()` method to retrieve RDI instance information
- Implements `selectPipeline()` method to automatically select the first available pipeline
- Adds support for v2-specific endpoints (`/api/v2/info`, `/api/v2/pipelines`)

#### 2. **Smart Client Factory** - Enhanced `RdiClientFactory` with version detection
- Attempts to create v2 client first by calling `getInfo()` endpoint
- If v2 API is available: initializes v2 client with `connect()` and `selectPipeline()`
- If v2 API is unavailable: gracefully falls back to v1 client
- Ensures backward compatibility with existing RDI instances

### **New Models & Types**
- `RdiInfo` - Model for RDI instance information (version)
- `GetInfoResponse` - Response type for `/api/v2/info` endpoint
- `GetPipelinesResponse` - Response type for `/api/v2/pipelines` endpoint
- `PipelineResponses` - Interface for pipeline data structure
- `RdiUrlV2` - Constants for v2 API endpoints

### **Documentation**
- Added comprehensive JSDoc comments for `getInfo()` and `selectPipeline()` methods
- Documented parameters, return types, exceptions, and usage examples

#### 1. **ApiV2RdiClient Tests** (`api.v2.rdi.client.spec.ts`)
- 10 tests covering `getInfo()` and `selectPipeline()` methods
- Tests for success cases, error handling, and edge cases (null/undefined/empty data)
- Validates proper transformation of API responses to model instances

#### 2. **RdiClientFactory Tests** (enhanced `rdi.client.factory.spec.ts`)
- 5 tests covering v2 client creation and v1 fallback scenarios
- Tests for version detection logic
- Tests for graceful degradation when v2 is unavailable
- Tests for error handling in both v2 and v1 flows

### **Unit Tests**
```bash
# Run all RDI tests
yarn test:api --testPathPattern=rdi

# Results: 176 tests passed ✓

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds initial RDI API v2 support with automatic detection and seamless fallback.
> 
> - **New client**: `ApiV2RdiClient` extends `ApiRdiClient` with `getInfo()` and `selectPipeline()` using v2 endpoints `RdiUrlV2.GetInfo` and `RdiUrlV2.GetPipelines`
> - **Factory update**: `RdiClientFactory` now probes v2 (`getInfo`), then connects and selects a pipeline; otherwise falls back to v1 client
> - **Models & exports**: New `RdiInfo` model; v2 response types (`GetInfoResponse`, `GetPipelinesResponse`); export additional exceptions; add `RdiUrlV2`
> - **Refactors**: Move v1 client imports under `api/v1/`; make v1 client logger `protected`
> - **Tests**: New specs for v2 client and factory covering success, fallback, and error paths
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit cfe7c7a97ceb35851f9aed19fa708069c0976a9d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

### CODE REVIEW

File: redisinsight/api/src/modules/rdi/client/api/v2/responses/pipeline.responses.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface PipelineResponses {
+  name: string;
+  active: boolean;
+  config: any; // todo: define
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/client/api/v2/responses/pipeline.responses.ts:4]: I believe you didn't forget these todos and left them intentionally like that, for the future iterations, right?

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   * await client.selectPipeline();
+   * // client.selectedPipeline is now set to the first available pipeline name
+   */
+  async selectPipeline(): Promise<void> {
+    try {
+      const { data } = await this.client.get<GetPipelinesResponse>(
+        RdiUrlV2.GetPipelines,
+      );
+
+      // todo: handle cases when no pipelines differently
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:66]: Do you have any other ideas what fallback mechanism we can have, when there are no pipelines available?

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      if (!data?.length) {
+        throw new RdiPipelineInternalServerErrorException(
+          'Unable to select pipeline',
+        );
+      }
+
+      this.selectedPipeline = data[0].name;
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:73]: Question: Is the name of the pipeline the best identifier, and do we have something like an ID?

File: redisinsight/api/src/modules/rdi/providers/rdi.client.factory.spec.ts [hunk 4]
```diff
@@ -27,31 +30,151 @@ describe('RdiClientFactory', () => {
...
-          access_token: mockedAccessToken,
-        },
+    describe('v2 client creation', () => {
+      it('should create v2 client when getInfo succeeds', async () => {
+        const mockedAccessToken = sign(
+          { exp: Math.trunc(Date.now() / 1000) + 3600 },
+          'test',
+        );
+        const mockInfoResponse = { version: '2.0.1' };
+        const mockPipelinesResponse = [
```
Comment (valkirilov) [inline redisinsight/api/src/modules/rdi/providers/rdi.client.factory.spec.ts:40]: Does it make sense to type-hint this one, or even add a factory so we can reuse it easily in many places? 

Or you left it untyped purposefully, and you don't think we'll benefit from the factory pattern at all.

File: redisinsight/api/src/modules/rdi/client/api/v2/responses/pipeline.responses.ts (hunk 1 above)
Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/responses/pipeline.responses.ts:4]: correct. I will appear at some point in the future PRs

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts (hunk 2 above)
Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:66]: It sounds like not realistic case for v2 but fallback might be to use v1. but at some point v1 will be removed... So I would say that we might want to handle it in some specific way if needed. this is more like a note to pay attention on it.

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts (hunk 3 above)
Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:73]: It is what we have by design in v2. all requests must include pipeline name

File: redisinsight/api/src/modules/rdi/providers/rdi.client.factory.spec.ts (hunk 4 above)
Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/providers/rdi.client.factory.spec.ts:40]: it makes perfect sense

Comment (ArtemHoruzhenko) [reply redisinsight/api/src/modules/rdi/providers/rdi.client.factory.spec.ts:40]: But unfortunately we don't have it for api. I would like to set it up in scope of a different task

File: redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts (hunk 3 above)
Comment (valkirilov) [reply redisinsight/api/src/modules/rdi/client/api/v2/api.v2.rdi.client.ts:73]: Ok then, I assume it's unique. Thanks :)

**Review by valkirilov — APPROVED** [review]
